include/swift/Runtime/Mutex.h

```diff
@@ -186,121 +114,593 @@ class Mutex {
   /// - The behavior is undefined if the calling thread already owns
   ///   the mutex (likely a deadlock)?
   /// - Does not throw exceptions but will halt on error (fatalError).
-  bool try_lock() { return Impl.try_lock(); }
+  bool try_lock() { return MutexPlatformHelper::try_lock(Handle); }
 
-public:
   /// Releases lock, waits on supplied condition, and relocks before returning.
   ///
-  /// Precondition: Mutex locked by this thread, undefined otherwise.
-  void wait(Condition &condition) { Impl.wait(condition); }
+  /// Precondition: Mutex held by this thread, undefined otherwise.
+  void wait(ConditionVariable &condition) {
+    ConditionPlatformHelper::wait(condition.Handle, Handle);
+  }
 
```

Comment:
I had moved it over when I had wait also available on ScopedLock, etc. It makes more sense to move it back now.